### PR TITLE
Update README and CHANGELOG for upcoming release

### PR DESCRIPTION
**Pull request by aggarw13**
Update README and CHANGELOG for upcoming release.

### CODE REVIEW

**Review by archigup — DISMISSED** [review]

File: README.md [hunk 1]
```diff
@@ -218,6 +226,14 @@ For example, a second release in June 2021 would be 202106.01. Although the SDK
 
 All of the released versions of the C-SDK libraries are available as git tags. For example, the last release of the v3 SDK version is available at [tag 3.1.2](https://github.com/aws/aws-iot-device-sdk-embedded-C/tree/v3.1.2).
 
+### 202108.00
+
+<!-- [API documentation of 202108.00 release](). -->
+
+This release introduces the refactored [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk)** library and the new [AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk) library.
```
Comment (archigup) [inline README.md:233]:
```suggestion
This release introduces the refactored [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk) library and the new [AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk) library.
```

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,6 +1,13 @@
...
+## 202108.00 (August 2021)
+
+### Major Changes
+- This release introduces the **[AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk)** library and **[AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk)** library.
+These libraries have gone through code quality checks including verification that no function has a [GNU Complexity](https://www.gnu.org/software/complexity/manual/complexity.html) score greater than 8, checks against deviations from the mandatory rules in the [MISRA coding standard](https://www.misra.org.uk), static code analysis from [Coverity statical analysis](https://scan.coverity.com/) and validation of memory safety through the [CBMC automated reasoning tool](http://www.cs.cmu.edu/~modelcheck/cbmc/).
+- [#1692](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1692) Update the existing HTTP S3 upload and download demos to obtain temporary credentials from [AWS IoT Credential Provider](https://docs.aws.amazon.com/iot/latest/developerguide/authorizing-direct-aws.html) and use the the new **AWS SigV4 library**  to sign HTTP requests with those credentials for authenticating with AWS S3 service.
```
Comment (gshvang) [inline CHANGELOG.md:8]: Only HTTP s3 download demos has been updated for this release.

File: README.md [hunk 3]
```diff
@@ -149,6 +151,12 @@ The [AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-a
 
 <!-- See memory requirements for the latest release [here](). -->
 
+#### AWS SigV4
+
+The [AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk) library enables you to sign HTTP requests with [Signature Version Signing Process](https://docs.aws.amazon.com/general/latest/gr/signature-version-4.html). Signature Version 4 (SigV4) is the process to add authentication information to HTTP requests to AWS services. For security, most requests to AWS must be signed with an access key. The access key consists of an access key ID and secret access key.
```
Comment (gshvang) [inline README.md:156]: The [AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk) library enables you to sign HTTP requests with [Signature Version 4 Signing Process]

**Review by gshvang — DISMISSED** [review]

**Review by yourslab — DISMISSED** [review]

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,6 +1,13 @@
...
+- [#1692](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1692) Update the existing HTTP S3 download demo to obtain temporary credentials from [AWS IoT Credential Provider](https://docs.aws.amazon.com/iot/latest/developerguide/authorizing-direct-aws.html) and use the the new **AWS SigV4 library**  to sign HTTP requests with those credentials for authenticating with AWS S3 service.
+- [#1688](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1688) Add Demo to showcase use of **AWS IoT Fleet Provisioning** library using the ["Provisioning by Claim"](https://docs.aws.amazon.com/iot/latest/developerguide/provision-wo-cert.html#claim-based) workflow of the Fleet Provisioning feature of AWS IoT Core.
+- [#1680](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1670) Add new transport interface implementation using **MbedTLS** (for TLS stack) and **corePKCS11** (for credential management).
```
Comment (tanmoysen) [inline CHANGELOG.md:10]: Mbed TLS

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,6 +1,13 @@
 # Changelog for AWS IoT Device SDK for Embedded C
 
-## main
+## 202108.00 (August 2021)
+
+### Major Changes
+- This release introduces the **[AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk)** library and **[AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk)** library.
```
Comment (tanmoysen) [inline CHANGELOG.md:6]: introduces a **refactored** AWS IoT Fleet Provisioning ....

Comment (aggarw13) [reply CHANGELOG.md:6]:
```suggestion
- This release introduces refactored the **[AWS IoT Fleet Provisioning](https://github.com/aws/fleet-provisioning-for-aws-iot-embedded-sdk)** library and **[AWS SigV4](https://github.com/aws/SigV4-for-AWS-IoT-embedded-sdk)** library.
```

File: CHANGELOG.md (hunk 4 above)
Comment (aggarw13) [reply CHANGELOG.md:10]:
```suggestion
- [#1680](https://github.com/aws/aws-iot-device-sdk-embedded-C/pull/1670) Add new transport interface implementation using **Mbed TLS** (for TLS stack) and **corePKCS11** (for credential management).
```

**Review by gshvang — APPROVED** [review]

**Review by abhidixi11 — APPROVED** [review]

**Pull request merged**